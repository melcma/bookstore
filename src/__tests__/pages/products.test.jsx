import { render, screen } from "@testing-library/react";
import ProductsPage, { getServerSideProps } from "../../pages/products";
import "@testing-library/jest-dom";

describe("ProductsPage", () => {
  it("renders a heading", () => {
    render(<ProductsPage />);

    const heading = screen.getByRole("heading", {
      name: /Products Page/i,
    });

    expect(heading).toBeInTheDocument();
  });

  it("renders products page unchanged", () => {
    const { container } = render(<ProductsPage />);
    expect(container).toMatchSnapshot();
  });

  it("renders products props", async () => {
    const response = await getServerSideProps();
    const products = response.props.products.edges;
    expect(products).not.toHaveLength(0);
  });

  it("renders no products when service fails", async () => {
    jest.mock("../../shopify/products");
  });
});
