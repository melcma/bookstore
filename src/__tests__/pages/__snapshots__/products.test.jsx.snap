// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsPage renders products page unchanged 1`] = `
<div>
  <main>
    <nav
      class="nav"
    >
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/products"
      >
        Shop
      </a>
      <a
        href="/cart"
      >
        Cart
      </a>
    </nav>
    <h1
      style="text-align: center;"
    >
      Products Page
    </h1>
    <section
      style="display: grid; grid-template-columns: repeat(3,1fr); gap: 1.5rem; margin-top: 20px;"
    />
  </main>
</div>
`;
